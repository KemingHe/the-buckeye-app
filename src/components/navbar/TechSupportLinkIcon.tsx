"use client";

import { QuestionMarkCircleIcon } from "@heroicons/react/24/outline";
import Link from "next/link";

// import { TECH_SUPPORT_ROUTE } from "@constants/routeConstants";

const TECH_SUPPORT_ROUTE: string = "/tech-support";

export default function TechSupportLinkIcon(): JSX.Element {
  return (
    <Link
      className="relative rounded-full bg-base-100 p-1 text-base-content hover:text-primary focus:text-primary focus:outline-none focus:ring-2 focus:ring-primary focus:ring-offset-2"
      href={TECH_SUPPORT_ROUTE}
    >
      <span className="absolute -inset-1.5" />
      <span className="sr-only">Link to the technical support page</span>
      <QuestionMarkCircleIcon aria-hidden="true" className="size-6" />
    </Link>
  );
}
