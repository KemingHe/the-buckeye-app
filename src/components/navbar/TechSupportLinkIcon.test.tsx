import { type RenderResult, render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import TechSupportLinkIcon from "./TechSupportLinkIcon";

describe("TechSupportLinkIcon component", () => {
  it("renders a link icon with the correct href and sr-only text", () => {
    render(<TechSupportLinkIcon />);

    const elementByRole: HTMLElement = screen.getByRole("link");
    expect(elementByRole).toBeVisible();
    expect(elementByRole).toHaveAttribute("href", "/tech-support");

    const elementBySR: HTMLElement = screen.getByText(
      "Link to the technical support page"
    );
    expect(elementBySR).toBeVisible();
  });

  it("matches the snapshot", () => {
    const { asFragment }: RenderResult = render(<TechSupportLinkIcon />);
    expect(asFragment()).toMatchSnapshot();
  });
});
