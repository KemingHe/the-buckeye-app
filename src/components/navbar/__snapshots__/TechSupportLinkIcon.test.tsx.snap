// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TechSupportLinkIcon component > matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="relative rounded-full bg-base-100 p-1 text-base-content hover:text-primary focus:text-primary focus:outline-none focus:ring-2 focus:ring-primary focus:ring-offset-2"
    href="/tech-support"
  >
    <span
      class="absolute -inset-1.5"
    />
    <span
      class="sr-only"
    >
      Link to the technical support page
    </span>
    <svg
      aria-hidden="true"
      class="size-6"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.879 7.519c1.171-1.025 3.071-1.025 4.242 0 1.172 1.025 1.172 2.687 0 3.712-.203.179-.43.326-.67.442-.745.361-1.45.999-1.45 1.827v.75M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9 5.25h.008v.008H12v-.008Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </a>
</DocumentFragment>
`;
